src/amo/reducers/reviews.js

```diff
@@ -490,6 +490,7 @@ export default function reviewsReducer(
       if (reviewData) {
         const { addonId, addonSlug, userId } = reviewData;
         delete newState.byId[reviewId];
+        delete newState.view[reviewId];
```

Comment:
Well, huh. I learned today that you can't use `delete` in a reducer.

Here's how to break the app:
- Submit a review
- Go to the review listing page
- Delete the review
- Open the Redux Devtool
- Jump back to the state triggered by the action `DELETE_ADDON_REVIEW`

The app is now broken (empty page, error in the console) because `delete` erased objects that were stored in previous states. Here's how to fix it (we need to always do this, so I filed https://github.com/mozilla/addons-frontend/issues/6102):

```diff
diff --git a/src/amo/reducers/reviews.js b/src/amo/reducers/reviews.js
index 240b26304..aadc7cb82 100644
--- a/src/amo/reducers/reviews.js
+++ b/src/amo/reducers/reviews.js
@@ -489,10 +489,16 @@ export default function reviewsReducer(
 
       if (reviewData) {
         const { addonId, addonSlug, userId } = reviewData;
-        delete newState.byId[reviewId];
-        delete newState.view[reviewId];
         return {
           ...newState,
+          byId: {
+            ...newState.byId,
+            [reviewId]: undefined,
+          },
+          view: {
+            ...newState.view,
+            [reviewId]: undefined,
+          },
           byAddon: {
             ...newState.byAddon,
             [addonSlug]: undefined,
```

I know that the only broken piece here is the Redux Devtool but I'd like to preserve that functionality. In this case, it's really important because I wanted to see what the loading state looks like. Without Redux time travel, it's hard to "pause" the app in a loading state.

Reply:
@kumar303 Did you try it with this patch? Because even with this patch the Redux devtools are still broken after I delete a review. I think it's because of the way the reducer works. I can look into it further, but I wanted to check whether implementing this solution actually fixed the redux devtools for you, or if you just assumed it would.